cli/sub_command.go

```diff
@@ -422,9 +418,58 @@ func (c *subCmd) createMsgHandler(hcfg msgHandlerState, nc *nats.Conn, subjMu *s
 	}
 }
 
-func (c *subCmd) shouldIgnore(m *nats.Msg, ignoreSubjects []string) bool {
+func (c *subCmd) createJestreamMsgHandler(hcfg msgHandlerState, nc *nats.Conn, subjMu *sync.Mutex, consumers *[]jetstream.ConsumeContext, ignoreSubjects []string, subjectReportMap map[string]int64, subjectBytesReportMap map[string]int64, matchMap map[string]*nats.Msg, t *time.Timer, dump bool) jetstream.MessageHandler {
```

Comment:
var here `hcfg` seems a bit like handler config, can we perhaps pass some of these things in via that instead?  

Otherwise at least should comment what this stuff all does, we'll be hating ourselves in the future

otoh the other also have these giant arg lists so probably just comment it